### DISCUSSION TITLE
[Plugin API] Allow plugins to control guest spawning

### DISCUSSION THREAD

**Discussion by kendfrey** [OpenRCT2/OpenRCT2#22848] [Feature requests] [answered]
## Story
I want to write a plugin that modifies the guest spawning mechanics. Currently the plugin API marks guests spawned with `createEntity` as `isInPark = true` which means they will not walk to the entrance and enter the park. Their properties are also set to 0 instead of the scenario defaults (cash, happiness, etc.)

## Proposal
Add a new function `map.spawnGuest(): Guest;`. This would spawn a guest in the same way as happens naturally or by the "Large tram" cheat.

## Nice to have
Add a flag to `GameMap` which disables natural guest generation, so that the plugin can take full control. This can be emulated by subscribing to `guest.generation` and removing naturally spawned guests, so it's not a requirement.

**Comment by J-PIE-314** [comment, marked as answer]
PR #22883 was created to add this functionality and has been merged.
